tests/test_models.py

```diff
@@ -0,0 +1,69 @@
+from magpie.models import UserStatuses
+from tests import runner
+
+
+@runner.MAGPIE_TEST_UTILS
+def test_user_status_value_getter():
+    assert UserStatuses.OK.value in UserStatuses.allowed()
+    assert UserStatuses.Pending.value in UserStatuses.allowed()
+    assert UserStatuses.WebhookError.value in UserStatuses.allowed()
+    assert str(UserStatuses.OK.value) in UserStatuses.allowed()
+    assert str(UserStatuses.Pending.value) in UserStatuses.allowed()
+    assert str(UserStatuses.WebhookError.value) in UserStatuses.allowed()
+    assert UserStatuses.OK.name in UserStatuses.allowed()
+    assert UserStatuses.Pending.name in UserStatuses.allowed()
+    assert UserStatuses.WebhookError.name in UserStatuses.allowed()
+
+    assert UserStatuses.OK in UserStatuses.all()
+    assert UserStatuses.Pending in UserStatuses.all()
+    assert UserStatuses.WebhookError in UserStatuses.all()
+    assert UserStatuses.OK | UserStatuses.Pending | UserStatuses.WebhookError == UserStatuses.all()
+    assert UserStatuses.get(1) is UserStatuses.OK
+    assert UserStatuses.get("ok") is UserStatuses.OK
+    assert UserStatuses.get("ok,PENDING") == UserStatuses.OK | UserStatuses.Pending
+    assert UserStatuses.get(["ok", "PENDING", "OK"]) == UserStatuses.OK | UserStatuses.Pending
+    assert UserStatuses.get("all") == UserStatuses.all()
+    assert UserStatuses.get(None) is None
+
+
+@runner.MAGPIE_TEST_UTILS
+def test_user_status_combinations():
+    # below order by value is important, sorted from highest to lowest bit (HB -> LB)
+    # this is the resolved order, no matter the way they are combined using OR operator '|' (enum impl and string repr)
+    test_statuses = [UserStatuses.Pending, UserStatuses.WebhookError, UserStatuses.OK]
+    flip_statuses = list(reversed(test_statuses))
+
+    # list() of class by itself returns all elements (same as .all())
+    # in this case, the member iterator is called, which prefers enum's impl to list by member definition
+    # this makes it reversed than the above HB -> LB in this case
+    assert list(UserStatuses) == flip_statuses
+    for idx, status in enumerate(reversed(UserStatuses)):
+        assert test_statuses[idx] is status
+
+    idx = None
+
+    # iterate items in random order merged by OR, using all items
+    merge_status = UserStatuses.Pending | UserStatuses.OK | UserStatuses.WebhookError
+    assert len(merge_status) == len(test_statuses)
```

Comment:
It looks that the len function is tested :), so it must work.
Will check it again!